vscode-sbcl-clpm.md

```diff
@@ -0,0 +1,472 @@
+           :depends-on ("experiment/src/app")
+           :description ""
+           :in-order-to ((test-op (load-op "experiment/test/all")))
+           :perform (test-op (o c) (symbol-call :test/all :test-suite)))
+
+         (defsystem "experiment/test"
+           :depends-on ("experiment/test/all"))
+
+         (register-system-packages "experiment/src/app" '(:app))
+         (register-system-packages "experiment/test/all" '(:test/all))
+         ~~~
+4. Set up a clpm bundle in your terminal
+   1. Start up your lisp repl
+      ```bash
+      rlwrap sbcl
+      ```
+   2. Init/install your clpm bundle, switch to your project context, and
+      load it.  **It will drop into a debugger on some of these steps,
+      if it does, just choose option 0 and hit enter.**  The non TL;DR;
+      version explains what is going on if you're curious.  It you screw
+      up the paths on these, hitting 0 will do nothing for you.  If you
+      find you're in an unrecoverable loop, it is often caused by a
+      fat-fingered path.
+      ~~~lisp
+      (clpm-client:bundle-init #p"/Users/me/Desktop/experiment/clpmfile" :asds `("experiment.asd"))
+      (clpm-client:install :context #p"/Users/me/Desktop/experiment/clpmfile")
+      (clpm-client:activate-context #p"/Users/me/Desktop/experiment/clpmfile" :activate-asdf-integration t)
+      (asdf:load-system :experiment)
+      ~~~
+5. Run your tests
+   ~~~lisp
+   (asdf:test-system :experiment)
+   ~~~
+6. Set up swank and hook it to vscode
+   1. In your clpmfile, add a new line that looks like this
+      ~~~lisp
+      (:system :swank)
+      ~~~
+   2. Your clpmfile should look something like this
+      ~~~lisp
+      ;;; -*- Mode: common-lisp; -*-
+      (:api-version "0.3")
+
+      (:source "quicklisp"
+               :url "https://quicklisp.common-lisp-project-index.org/"
+               :type :ql-clpi)
+
+      (:asd "experiment.asd")
+
+      ;; Dev Dependencies
+      (:system :swank) ;; Swank for that dank repl dev
+      ~~~
+   3. Have clpm download swank, load it with asdf, and start it up
+      ```bash
+      (clpm-client:install :context #p"/Users/me/Desktop/experiment/clpmfile")
+      (asdf:load-system :swank)
+      (swank:create-server)
+      ```
+   4. Attach vscode to the repl from inside of vscode and get to work.
+      - `View/Command Palette`
+      - `Alive: Attach to REPL`
+   5. To do this faster after it's all set up, you can start SBCL with
+      clpm using this command and have it automatically load swank and
+      start a repl and automatically install missing dependencies when
+      it encounters them.
+      ~~~sh
+      clpm exec --context=experiment -- sbcl \
+      --eval '(setf clpm-client:*asdf-system-not-found-behavior* :install)' \
+      --eval '(setf clpm-client:*context-diff-approval-method* t)' \
+      --eval '(asdf:load-system :swank)' \
+      --eval '(swank:create-server)'
+      ~~~
+
+### The Longer Version
+#### Things To Keep in Mind
+- Read
```

Comment:
A reminder about the components might be useful, but I find this section a bit long for a recipe about VSCode. It would do great on a blog post, less here.

Reply:
Fixed.